### PR TITLE
Parallel coverage

### PR DESCRIPTION
**Pull request by ailisp**
Some time we get port already in use, and it's likely the expensive test lock not properly protect expensive tests running in parallel.: https://github.com/nearprotocol/nearcore/issues/1784

Update: This also seems to be a 3x speedup when running locally:
python3 scripts/parallel_run_tests.py  7.70s user 3.19s system 7% cpu 2:30.50 total
cargo test --workspace  548.37s user 21.83s system 113% cpu 8:20.56 total

### CODE REVIEW

**Comment by ailisp** [comment]
Surprised that buidl-x gitlab-runner doesn't in docker group. Added them in all runners, hopefully pass in next commit

File: scripts/parallel_run_tests.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    if os.path.isfile(f):
+        os.remove(f)
+
+# Build tests
+p = subprocess.run(['cargo', 'test', '--workspace', '--no-run'])
+if p.returncode != 0:
+    os._exit(p.returncode)
+binaries = []
+for f in glob.glob(f'{target_debug}/*'):
+    fname = f.split('/')[-1]
```
Comment (nearmax) [inline scripts/parallel_run_tests.py]: Nit. Please use `os.path` to extract filename from path, for safety reasons.

File: scripts/parallel_run_tests.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    if os.path.isfile(f) and fname != 'near' and not f.endswith('.d') and not fname.startswith('test_regression'):
```
Comment (nearmax) [inline scripts/parallel_run_tests.py]: Nit. Use https://docs.python.org/3/library/os.path.html#os.path.splitext

**Comment by ailisp** [comment]
wait for a few more runs to see if flaky get better.

File: scripts/parallel_run_tests.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            binary = future_to_binary[future]
+            result = future.result()
+            print(f'========= test {binary}')
+            print('========= stdout:')
+            print(result[1])
+            print('========= stderr:')
+            print(result[2])
+            if result[0] != 0:
+                print(f'========= test {binary} exit code {result[0]} cause build fail')
+                os._exit(result[0])
```
Comment (nearmax) [inline scripts/parallel_run_tests.py]: We should also make sure the subprocesses are terminated when the main process is terminated.

**Comment by ailisp** [comment]
Thanks for everyone's help, run-parallel-test seems quite reliable locally. Run five times and no test fails. For coverage still wip

File: scripts/parallel_run_tests.py (hunk 3 above)
Comment (ailisp) [reply scripts/parallel_run_tests.py]: Good point. Was adding this but @SkidanovAlex said it's better to run all tests even some test fails. so don't need exit early and no cleanup now

**Comment by ailisp** [comment]
Parallel code coverage is finished too. Locally it takes 6 minutes. Please take a another look. @nearmax @bowenwang1996 @SkidanovAlex

**Comment by ailisp** [comment]
io seems not a bound according to iotop, they're rarely over 1% io of a <10M disk r/w. It's weird this test timeout.

**Comment by ailisp** [comment]
Tests still seem to volatile. Coverage is much faster and stable so only keep python parallel for now. Python testlib/parallel_run_test would be helpful locally, as it saves 50%-75% time

**Pull request merged**